src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -93,7 +93,11 @@ public static IEnumerable<VisualStudioInstance> QueryVisualStudioInstances()
         /// <returns>Instance of Visual Studio found and registered.</returns>
         public static VisualStudioInstance RegisterDefaults()
         {
-            var instance = GetInstances(VisualStudioInstanceQueryOptions.Default).FirstOrDefault();
+            IEnumerable<VisualStudioInstance> instances = GetInstances(VisualStudioInstanceQueryOptions.Default);
+            VisualStudioInstance instance = instances.Where(
+                inst => inst.Version.Major < Environment.Version.Major || (inst.Version.Major == Environment.Version.Major && inst.Version.Minor <= Environment.Version.Minor)
+                ).FirstOrDefault() ??
+                instances.FirstOrDefault();
```

Comment:
Do you think I should keep this as a fallback? It only ever would return an SDK from after the runtime, but "No instances of MSBuild could be detected" would be misleading.